### PR TITLE
Add document for the development process

### PR DESCRIPTION
**Pull request by eopXD**
We need explicit documentation for a standard of process that we can follow since there may be possible incompatible changes in the future. Also a general rule on how a pull request can be approved.

### CODE REVIEW

**Review by rofirrim — COMMENTED** [review]
Thanks @eopXD overall looks reasonable, a couple of minor comments.

File: development_process.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+=== Requirement in the next release
+
+For incompatible change that removes API-(s) from rvv-intrinsic-doc, the implementation of the change in the **next** release is expected to label the to-be-deleted intrinsics as **deprecated**. The compiler should implement **warning** for usage of the deprecated API and add hint to its resolution.
+
+For incompatible change that changes API semantic, the implementation of the change in the **next** release is expected to apply the semantic change with warnings to notify the user of such change.
+
+=== Requirement in the next-next release
+
+For incompatible change that removes API-(s) from rvv-intrinsic-doc, the implementation of the change in the **next-next** release is to remove the deprecated intrinsics labeled in the previous release. The compiler should implement **error** to hint for its resolution.
```
[the remark this answers is not in the document]
Comment (eopXD) [reply development_process.adoc:38]: We can either have
- Plan A: Compiler be aware of the removed intrinsic and provide hint for resolution in the next-next release, then be oblivious of the removed intrinsics in the next-next-next release.
- Plan B: Compiler be oblivious of the removed intrinsics in the next-next release.

I would prefer PlanA. Although GCC releases once per year, which would take us 3 release cycles to remove an intrinsic, but I think this is a more conservative approach that allows users to adapt the incompatible change through time.

[the remark this answers is not in the document]
Comment (rofirrim) [reply development_process.adoc:40]: I'm OK with either option, Plan A may take longer but makes next-next a bit more informative. Functionally next-next and next-next-next are the same in that they'd reject the code.

[the remark this answers is not in the document]
Comment (rofirrim) [reply development_process.adoc:40]: If we choose Plan A I guess we want to document that there is no specific requirement on diagnostics regarding the removed API for next-next-next.

File: development_process.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+The issue may be closed as un-related if it is not related to the designs or the development of the RVV intrinsics.
+
+== General rules to act on a Pull Request
+
+Apart from the criteria below, it is expected that all PR approaches to resolve the conflicts and concerns raised. When discussion comes to a deadlock, the active coordinator of this document shall attempt to clarify existing objections, coordinate between discussion participants and push the progress forward to either approving or rejecting the proposal.
+
+=== Pull request of API-related change
+
+To a pull request of API-related change, the change often come in chunks. However we don't need the **full implementation** before any change is approved and applied into the document. A pull request of such requires a **Proof-of-Concept** in either GCC or LLVM.
```
[the remark this answers is not in the document]
Comment (eopXD) [reply development_process.adoc:18]: A paragraph is added to clarify this, marking this as resolved.

File: development_process.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (eopXD) [reply development_process.adoc:40]: Added requirement for next-next-next.

**Comment by eopXD** [comment]
Change:
- Fix gramatical error
- Add requirement in the next-next-next release for applying incompatible changes.

**Review by rofirrim — APPROVED** [review]
The process as described LGTM.

**Comment by eopXD** [comment]
Dropping this PR. After discussion in the open meeting, how the LLVM deal with existing implemented intrinsic will go back to the LLVM community itself. The RVV intrinsic repository will focus on the specification and are not responsible in defining such process.

**Pull request closed without merging**